src/components/RegisterForm/RegisterForm.tsx

```diff
@@ -15,9 +16,33 @@ export function RegisterForm() {
   const navigate = useNavigate();
   const { mutate, isLoading, isSuccess, isError } = useCreateUserMutation();
 
-  const onFormSubmit = (values: CreateUserType) => {
+  function isRecordWithAllFields<T extends object>(record: T | null | undefined, keys: Array<keyof T>): record is Required<T> {
+    if (record === undefined || record === null) {
+      return false;
+    }
+
+    for (const key of keys) {
+      if (!(key in record)) {
+        return false;
+      }
+    }
+
+    return true;
+  }
```

Comment:
This is a helper function. Please put this inside the appropriate directory
https://github.com/dorm-social-room-manager/dsrm-frontend/wiki/Project-structure#srccommonutils